src/database/schemamanager.h

```diff
@@ -26,8 +29,18 @@ class SchemaManager {
         return m_currentVersion;
     }
 
+    /// Check if the current version is backwards compatible with the given targetVersion.
+    ///
+    /// If the current version has no backwards compatibility info, it is
+    /// assumed that it is only compatible with itself.
+    ///
+    /// Returns true if the targetVersion is greater or equal to the
+    /// minimum compatible version of the current schema
     bool isBackwardsCompatibleWithVersion(int targetVersion) const;
 
+    /// Tries to upgrade the database schema to the given targetVersion.
+    /// First checks for version compatibility and whether the current version
```

Comment:
Repeating the internal implementation of a function as documentation is an ant-pattern. This does not belong here. It will get inconsistent when changing the implementation.

Reply:
And I don't need the signature of the function repeated in someone's own word instead of C++.

Reply:
What would you suggest then? I think it does give a nice high-level overview. If the version compatibility check would be removed, changing the documentation would also be appropriate, as this is a sever functional difference, no?

Reply:
Concise documentation without additional redundancy. A good function name and signature should speak for itself.